test/main/externalData/inventory/processor.test.ts

```diff
@@ -0,0 +1,236 @@
+
+    expect(store.setInventory).toHaveBeenCalledWith(account.toLowerCase(), {
+      '0x1': {
+        ...newInventory['0x1'],
+        items: {
+          '1': {
+            name: 'name'
+          },
+          '2': {}
+        }
+      }
+    })
+  })
+
+  it('removes data of items no longer owned after inventory update', () => {
+    // Create an existing inventory with two items in one collection
+    mockExistingInventory = {
+      '0x1': Collection({
+        '1': { name: 'name' },
+        '2': { name: 'otherName' }
+      })
+    }
+
+    // Create a new inventory with only one of the items from the existing inventory
+    const newInventory = {
+      '0x1': Collection({
+        '1': {} // Still owned item
+        // Item '2' is not included, so it's no longer owned
+      })
+    }
+
+    updateCollections(account, newInventory)
+
+    expect(store.setInventory).toHaveBeenCalledWith(account.toLowerCase(), {
+      '0x1': {
+        ...newInventory['0x1'],
+        items: {
+          '1': { name: 'name' } // Still owned item with preserved data
+          // Item '2' data should not be present
+        }
+      }
+    })
+  })
+})
+
+describe('updateItems', () => {
+  // 1. Update an existing item in the inventory
+  it('updates the inventory asset data for existing items', () => {
+    const items = [
+      {
+        contract: '0x1',
+        tokenId: '1',
+        name: 'newName'
+      },
+      {
+        contract: '0x1',
+        tokenId: '2',
+        name: 'name2'
+      }
+    ]
+
+    mockExistingInventory = {
+      '0x1': Collection({
+        '1': { contract: '0x1', tokenId: '1', name: 'name' },
+        '2': {}
+      })
+    }
+
+    updateItems(account, items)
+
+    expect(store.setInventoryAsset).toHaveBeenNthCalledWith(1, account.toLowerCase(), '0x1', '1', {
+      ...items[0],
+      name: 'newName'
+    })
+
+    expect(store.setInventoryAsset).toHaveBeenNthCalledWith(2, account.toLowerCase(), '0x1', '2', items[1])
+  })
+
+  // 2. Try to update a non-existent item
+  it('logs a warning and does not update when the item does not exist in the inventory', () => {
+    const items = [
+      {
+        contract: '0x1',
+        tokenId: '2',
+        name: 'updatedName'
+      }
+    ]
+    mockExistingInventory = {
+      '0x1': Collection({
+        '1': { name: 'oldName' }
+      })
+    }
+
+    updateItems(account, items)
+
+    expect(log.warn).toHaveBeenCalled()
+    expect(store.setInventoryAsset).not.toHaveBeenCalled()
+  })
+
+  // 3. Try to update an item in a non-existent collection
+  it('logs a warning and does not update when the collection does not exist in the inventory', () => {
+    const items = [
+      {
+        contract: '0x2',
+        tokenId: '1',
+        name: 'updatedName'
+      }
+    ]
+
+    mockExistingInventory = {
+      '0x1': Collection({
+        '1': { name: 'oldName' }
+      })
+    }
+
+    updateItems(account, items)
+
+    expect(log.warn).toHaveBeenCalled()
```

Comment:
I dont think we need to verify logging, it makes the tests a little brittle as we may want to remove this logging or change the log level, etc